(DOCSP-26742): Flutter custom user data (#2332)
mongodb/docs-realm: source/sdk/flutter/users/access-token.txt

.. _flutter-access-token:

=======================================
Get the User Access Token - Flutter SDK
=======================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

> Every User object contains a JWT token which you can use to access Atlas App Services.

Reviewer: ```suggestion
Every User object contains a JWT token that you can use to access Atlas App Services.
```

> You can use the access token to query the Atlas GraphQL API from your client application

Reviewer: This sentence could be reworked or broken up. There are two ideas here: "you can use the token to do what" and "using a GraphQL client, like...". You can use the token with any GraphQL client to query the Atlas GraphQL API.

> You can use the access token to query the Atlas GraphQL API from your client application
> using any GraphQL client, like `graphql_flutter <https://pub.dev/packages/graphql_flutter>`__.

Author: ```suggestion
You can use the access token to query the Atlas GraphQL API from your client application.
Use any GraphQL client to query the Atlas GraphQL API, such as `graphql_flutter <https://pub.dev/packages/graphql_flutter>`__.
```

To learn more about setting up and querying the Atlas GraphQL API,
refer to :ref:`graphql-api` in the App Services documentation.

Retrieve the Access Token
-------------------------

You can get the access token with the :flutter-sdk:`User.accessToken
<realm/User/accessToken.html>` property.

.. literalinclude:: /examples/generated/flutter/access_token_test.snippet.get-access-token.dart
   :language: dart

Refresh the Access Token
------------------------

The access token expires 30 minutes after a user logs in. It does not refresh automatically.

> You can refresh it with :flutter-sdk:`User.refreshCustomData() <realm/User/refreshCustomData.html>`.

Reviewer: ```suggestion
Refresh it with :flutter-sdk:`User.refreshCustomData() <realm/User/refreshCustomData.html>`.
```

.. literalinclude:: /examples/generated/flutter/access_token_test.snippet.refresh-access-token.dart
   :language: dart

> You can also periodically refresh the access token to make sure it's always up to date

Reviewer: ```suggestion
You can also periodically refresh the access token 
```

with `Timer.periodic() <https://api.dart.dev/be/180360/dart-async/Timer-class.html>`__
from the ``dart:async`` library. Wrap the call to ``User.refreshCustomData()``
with the timer's callback function.

.. literalinclude:: /examples/generated/flutter/access_token_test.snippet.periodic-refresh-access-token.dart
   :language: dart

> 
> 
> 
> 
> 

Author: ```suggestion
```
